Source/com/drew/metadata/iptc/IptcReader.java

```diff
@@ -114,8 +114,8 @@ public void extract(@NotNull final SequentialReader reader, @NotNull final Metad
                 return;
             }
 
-            // we need at least five bytes left to read a tag
-            if (offset + 5 >= length) {
+            // we need at least four bytes left to read a tag
+            if (offset + 4 >= length) {
```

Comment:
Actually I think the comment is correct. You need at least 5 bytes:

- 1 for dir type
- 1 for tag type
- 2 for byte count
- 1+ for contents

I don't think it's valid to have zero bytes of content.

Do you have an image that reproduces this behaviour? I can't quite make out the problem here.